src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -350,6 +363,34 @@ public void postPatronAccountInstanceHoldByIdAndInstanceId(String id,
     }
   }
 
+  @Override
+  public void getPatronAccountInstanceAllowedServicePointsByIdAndInstanceId(String requesterId,
+    String instanceId, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    var httpClient = HttpClientFactory.getHttpClient(vertxContext.owner());
+    String url = format(CIRCULATION_REQUESTS_ALLOWED_SERVICE_POINTS, requesterId, instanceId);
+
+    try {
+      httpClient.get(url, okapiHeaders)
+        .thenApply(ResponseInterpreter::verifyAndExtractBody)
+        .thenAccept(body -> {
+          AllowedServicePoints allowedServicePoints = getAllowedServicePoints(body);
+          asyncResultHandler.handle(succeededFuture(
+            GetPatronAccountInstanceAllowedServicePointsByIdAndInstanceIdResponse
+              .respond200WithApplicationJson(allowedServicePoints)));
+        })
+        .exceptionally(throwable -> {
```

Comment:
I would suggest using .whenComplete for both cases

Reply:
done